frontend/src/pages/modelRegistry/screens/RegisterModel/RegisterModel.tsx

```diff
@@ -0,0 +1,335 @@
+                    Configure the model info you want to create.
+                  </Text>
+                </>
+              }
+            >
+              <FormGroup label="Model name" isRequired fieldId="model-name">
+                <TextInput
+                  isRequired
+                  type="text"
+                  id="model-name"
+                  name="model-name"
+                  value={modelName}
+                  onChange={(_e, value) => setData('modelName', value)}
+                />
+              </FormGroup>
+              <FormGroup label="Model description" fieldId="model-description">
+                <TextArea
+                  type="text"
+                  id="model-description"
+                  name="model-description"
+                  value={modelDescription}
+                  onChange={(_e, value) => setData('modelDescription', value)}
+                />
+              </FormGroup>
+            </FormSection>
+            <FormSection
+              title={
+                <>
+                  <Title headingLevel="h2">Version info</Title>
+                  <Text component="p" className="form-subtitle-text">
+                    Configure the version info for the run that you want to register.
+                  </Text>
+                </>
+              }
+            >
+              <FormGroup label="Version name" isRequired fieldId="version-name">
+                <TextInput
+                  isRequired
+                  type="text"
+                  id="version-name"
+                  name="version-name"
+                  value={versionName}
+                  onChange={(_e, value) => setData('versionName', value)}
+                />
+              </FormGroup>
+              <FormGroup label="Version description" fieldId="version-description">
+                <TextArea
+                  type="text"
+                  id="version-description"
+                  name="version-description"
+                  value={versionDescription}
+                  onChange={(_e, value) => setData('versionDescription', value)}
+                />
+              </FormGroup>
+              <FormGroup label="Source model format" fieldId="source-model-format">
+                <TextInput
+                  type="text"
+                  id="source-model-format"
+                  name="source-model-format"
+                  value={sourceModelFormat}
+                  onChange={(_e, value) => setData('sourceModelFormat', value)}
+                />
+              </FormGroup>
+            </FormSection>
+            <FormSection
+              title={
+                <>
+                  <Title headingLevel="h2">Model location</Title>
+                  <Text component="p" className="form-subtitle-text">
+                    Specify the model location by providing either the object storage details or the
+                    URI. You can autofill the location details from an existing data connection.
```

Comment:
Sorry, can you remove the "You can autofill the location details from an existing data connection" sentence? We shouldn't add it until that functionality exists. I'll make a note to add it back in https://issues.redhat.com/browse/RHOAIENG-9914.